_pytest/debugging.py

```diff
@@ -85,6 +85,17 @@ def _enter_pdb(node, excinfo, rep):
     # for not completely clear reasons.
     tw = node.config.pluginmanager.getplugin("terminalreporter")._tw
     tw.line()
+
+    captured_stdout = rep.capstdout
+    if len(captured_stdout) > 0:
+        tw.sep(">", "captured stdout")
+        tw.line(captured_stdout)
+
+    captured_stderr = rep.capstderr
+    if len(captured_stderr) > 0:
+        tw.sep(">", "captured stderr")
+        tw.line(captured_stderr)
+
```

Comment:
Please also output the captured logs (``rep.caplog``) and make sure that they are not output if live-logging is enabled (``-o log_cli=True -o log_level=DEBUG``)

Reply:
Note that ``rep.caplog`` was added in the features branch. So you can't use it if you merge your branch into master, but you can copy'n'paste the implemenation of BaseReport.caplog from https://github.com/pytest-dev/pytest/pull/3156/files#diff-e1a5c52f502faf325466018a5440c400R260.

Reply:
> Please also output the captured logs (rep.caplog) and make sure that they are not output if live-logging is enabled (-o log_cli=True -o log_level=DEBUG)

Why should we make a special case for living log here? I kind see the point, but we are dumping captured stdout/stderr at this point it follows that we should dump caplog then as well, regardless of live logging. 

> but you can copy'n'paste the implemenation of BaseReport.caplog from

If we end up doing this, please don't make `caplog` public: we don't want to introduce a new public API in a bug-fix release.

Reply:
maybe it would be better to merge this and open a new issue to implement the same thing for caplog, in the features branch

Reply:
@brianmaissy fair enough. @thisch do you agree?

Reply:
Ok. Let's do it like this

Reply:
Opened a new issue #3204